File: src/main/java/org/flighthub/Repository/IClientRepository.java
package org.flighthub.Repository;

import org.flighthub.Domain.Client;

import java.util.Optional;
import java.util.UUID;

public interface IClientRepository extends Repository<UUID, Client> {
    Optional<Client> findOne(UUID id);

    Iterable<Client> findAll();

    Client save(Client entity);

    Optional<Client> delete(UUID id);

    Optional<Client> update(Client entity);
}

File: Server - C#/persistence/IClientRepository.cs
using FlightHubC_.Domain;
using FlightHubC_.Repository;
using System.Collections.Generic;
using System;
namespace FlightHubC_.Repository
{
    public interface IClientRepository : IRepository<Guid, Client>
    {
        Client FindOne(Guid id);
        IEnumerable<Client> FindAll();
        Client Save(Client entity);
        Client Delete(Guid id);
        Client Update(Client entity);
    }
}